pulp_rpm/app/advisory.py

```diff
@@ -0,0 +1,206 @@
+from gettext import gettext as _
+from itertools import chain
+
+import hashlib
+
+import createrepo_c as cr
+
+from django.db import (
+    IntegrityError,
+    transaction,
+)
+from django.utils.dateparse import parse_datetime
+
+from pulpcore.plugin.models import (
+    Content,
+    RepositoryContent,
+)
+
+from pulp_rpm.app.exceptions import AdvisoryConflict
+
+
+def resolve_advisories(version, previous_version):
+    """
+    Decide which advisories to add to a repo version and which to remove, and adjust a repo version.
+
+    Args:
+        version (pulpcore.app.models.RepositoryVersion): current incomplete repository version
+        previous_version (pulpcore.app.models.RepositoryVersion):  a version preceding
+                                                                   the current incomplete one
+
+    """
+    content_pks_to_add = set()
+    content_pks_to_remove = set()
+    content_pks_to_exclude = set()  # exclude from the set of content which is being added
+
+    # identify conflicting advisories
+    current_advisories = version.content.filter(pulp_type='rpm.advisory')
+    added_advisories = current_advisories
+    advisory_conflicts = []
+    if previous_version:
+        previous_advisories = previous_version.content.filter(pulp_type='rpm.advisory')
+        previous_advisory_ids = set(previous_advisories.values_list('rpm_updaterecord__id',
+                                                                    flat=True))
```

Comment:
If you convert previous_advisories to UpdateRecords (see my other comment), you can simplify this:

```
previous_advisory_ids = set(previous_advisories.values_list('id', flat=True))
```